libr/anal/p/anal_riscv.c

```diff
@@ -58,9 +59,17 @@ static int riscv_op(RAnal *anal, RAnalOp *op, ut64 addr, const ut8 *data, int le
 	op->addr = addr;
 	op->type = R_ANAL_OP_TYPE_UNK;
 
-	if (len>=sizeof (word))
-		r_mem_copyendian ((void*)&word, (const void*)data, sizeof (word), true);
-	else r_mem_copyendian ((void*)&word, (const void*)data, 2, true); //sizeof (word), true);
+	if (len>=sizeof (word)) {
+		if (!anal->big_endian)
```

Comment:
this looks like spaguetty and using negative conditionals makes it harder to read